src/core_plugins/getting_started/public/lib/add_setup_work.js

```diff
@@ -11,6 +11,67 @@ import {
   CREATE_INDEX_PATTERN_ROUTE
 } from './constants';
 
+function handleExistingIndexPatternsScenario(indexPatterns, currentRoute, config) {
+  // If index patterns exist, we're not going to show the user the Getting Starte page.
+  // So we can show the chrome again at this point.
+  uiChrome.setVisible(true);
+
+  // The user need not see the Getting Started page, so opt them out of it
+  optOutOfGettingStarted();
+
+  // Some routes require a default index pattern to be present. If we're
+  // NOT on such a route, there's nothing more to do; send the user on their way
+  if (!currentRoute.requireDefaultIndex) {
+    return;
+  }
+
+  // Otherwise, check if we have a default index pattern
+  let defaultIndexPattern = config.get('defaultIndex');
```

Comment:
So the reason the two pieces of logic (getting started and default index patterns) have been combined is that both use `RouteSetupManager.addSetupWork(fn)` to do their job. In an early implementation I had the two independent of each other but I would see the "No default index pattern" notifier on the Getting Started page. This was happening because, AFAICT, there's no way of specifying the order of execution of the two `fn`s registered with `RouteSetupManager.addSetupWork(fn)`. So my solution to control the order was to combine the two pieces of logic into one `fn`.